### PR TITLE
Change method from `GET` to `POST` in API login requests

### LINKED ISSUE #2705: Change HTTP method to `POST` when using the API login endpoint
This issue is part of https://github.com/wazuh/wazuh/issues/12793.

The epic issue aims to change the API login endpoint method from `GET` to `POST`, as using `GET` as the method of a login endpoint is incorrect.

As it was said in https://github.com/wazuh/wazuh/issues/10203#issuecomment-929557731, both `GET /security/user/authenticate` and `POST /security/user/authenticate` endpoints will coexist and the deprecation of `GET /security/user/authenticate` will happen in a future major release. 

To avoid future problems when the actual login endpoint (GET) is deprecated, we should update all the references found in this repository:

- Occurrences of API_LOGIN_ENDPOINT

https://github.com/wazuh/wazuh-qa/blob/13db0b83da57ad8f4f112945e97ae96a60a8b503/deps/wazuh_testing/wazuh_testing/api.py#L22

https://github.com/wazuh/wazuh-qa/blob/13db0b83da57ad8f4f112945e97ae96a60a8b503/deps/wazuh_testing/wazuh_testing/api.py#L53-L65

https://github.com/wazuh/wazuh-qa/blob/13db0b83da57ad8f4f112945e97ae96a60a8b503/deps/wazuh_testing/wazuh_testing/api.py#L70

https://github.com/wazuh/wazuh-qa/blob/13db0b83da57ad8f4f112945e97ae96a60a8b503/tests/integration/test_api/test_config/test_request_timeout/test_request_timeout.py#L126

### PR DESCRIPTION
**Pull request by mcarmona99**
|Related issue|
|---|
|#2705|

## Description

This PR closes #2705.

Apart from the change listed in the related issue, I have added more changes to tests and tools using the login endpoint.

The related API tests will fail until https://github.com/wazuh/wazuh/issues/10203 is merged.

## Checks

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/tools/api_simulator.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def get_token(self):
         for _ in range(10):
             try:
                 self.logger.info('Trying to obtain API token')
-                response = requests.get(f'{self.base_url}{authenticate_url}', headers=basic_auth, verify=False)
+                response = requests.post(f'{self.base_url}{authenticate_url}', headers=basic_auth, verify=False)
```
Comment (vicferpoy) [inline deps/wazuh_testing/wazuh_testing/tools/api_simulator.py:95]:
```suggestion
                response = requests.post(f"{self.base_url}{authenticate_url}", headers=basic_auth, verify=False)
```

File: deps/wazuh_testing/wazuh_testing/tools/key_polling_api.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def insert_agent_manual(token, agent_id=None, agent_ip=None, agent_name=None, ag
 
 def read_token(new_token=False):
     def obtain_token():
-        response = requests.get(f'{BASE_URL}/security/user/authenticate', headers=login_headers(), verify=False)
+        response = requests.post(f'{BASE_URL}/security/user/authenticate', headers=login_headers(), verify=False)
```
Comment (vicferpoy) [inline deps/wazuh_testing/wazuh_testing/tools/key_polling_api.py:54]:
```suggestion
        response = requests.post(f"{BASE_URL}/security/user/authenticate", headers=login_headers(), verify=False)
```

**Review by vicferpoy — CHANGES_REQUESTED** [review]

**Review by vicferpoy — APPROVED** [review]
LGTM

**Review by damarisg — APPROVED** [review]
## Tests Results
| Test Path | Os/Type | Jenkins | Date | By | 
|--|--|--|--|--|
| test_api/  | Manager | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25980/)  [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25979/)  [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/25978/) | 03/05/2022 |  Seyla

Build package: https://ci.wazuh.info/job/Packages_builder/95111/

**Pull request merged**